### DISCUSSION TITLE
Recommended color map for NDWI

### DISCUSSION THREAD

**Discussion by neelduttahere** [cogeotiff/rio-tiler#595] [General]
Just like #90, what is the best colormap for NDWI? I can't find any suitable option and over that passing the custom colormap formula, `rio-tiler` Is unable to return a tile, in the region without data. That is, the edges of the image fail to load as they only have partial image data. Still searching for some way about this issue, but for now, I am willing to resort to the already available colormap options.

Thanks in advance

**Comment by vincentsarago** [comment]
>  what is the best colormap for NDWI?

Well it depends if you want a continuous color map or an interval colormap 🤷  This is quite subjective. We often let enduser select there own colormap.

> rio-tiler Is unable to return a tile, in the region without data. 

I'm not sure to get this. if the `tile` is within the dataset bounds, rio-tiler should return a tile, even if it's totally empty. I suggest to share a minimal reproducible example.

**Reply by neelduttahere** [reply]
Hey so I feel moving this discussion to [Titiler](https://github.com/developmentseed/titiler) makes more sense, as I faced this issue from Titiler. Though I am sharing some reproducible examples here as well:

Using this Sentinal-2 tile from Planetary -> https://planetarycomputer.microsoft.com/api/stac/v1/collections/sentinel-2-l2a/items/S2A_MSIL2A_20230402T053641_R005_T42QZJ_20230402T103026

Below are two Titiler request URLs for `/stac/tiles` using (B03-B08)/(B03+B08) expression (for NDWI)

1. Using default colormaps available in `rio-tiler`

https://{titiler-url}/stac/tiles/WebMercatorQuad/12/2877/1799?url=https://planetarycomputer.microsoft.com/api/stac/v1/collections/sentinel-2-l2a/items/S2A_MSIL2A_20230402T053641_R005_T42QZJ_20230402T103026&resampling_method=nearest&expression=%28B03-B08%29%2F%28B03%2BB08%29&rescale=-1%2C1&colormap_name=rplumbo

This request will return an image, but if you notice carefully, the bottom right corner of the tile has no data, which is expected considering the image returned was of the edge of the sentinel tile.

2. However, the same request is made with a minor modification, instead of using default colormaps, I am using a custom interval colormap -> 

https://{titiler}/stac/tiles/WebMercatorQuad/12/2877/1799?url=https://planetarycomputer.microsoft.com/api/stac/v1/collections/sentinel-2-l2a/items/S2A_MSIL2A_20230402T053641_R005_T42QZJ_20230402T103026&resampling_method=nearest&expression=%28B03-B08%29%2F%28B03%2BB08%29&colormap=[[[-1,-0.6],[255,%20255,%20255,%201]],[[-0.6,-0.3],[128,%20185,%20137,%201]],[[-0.3,0],[40,%20135,%2052,%201]],[[0,0.1],[7,%20127,%2066,%201]],[[0.1,0.2],[5,%20140,%20125,%201]],[[0.2,0.3],[3,%20153,%20187,%201]],[[0.3,0.6],[2,%20162,%20245,%201]],[[0.6,0.8],[0,%20120,%20212,%201]],[[0.8,1],[0,%2042,%20144,%201]]]

You will notice it returns a completely blank image, for the same location. Though this doesn't happen when the image request is in the middle region of the sentinel tile, as there are no sections in the middle of the tile that has `no-data`, a request like this where 99% of the satellite tile falls under the requested map tile from user's viewport, the 1% that has no data, returns a completely blank image as result, when custom-colormap is being used.

**Reply by vincentsarago** [reply]
I feel this is the same thing as https://github.com/developmentseed/titiler/discussions/588

**Reply by vincentsarago** [reply]
it also seems that you are using an old version of titiler (in new version you would have to add `asset_as_band=True` to use this URL. 

Sadly I cannot test this because I don't have a Tiler deployed in Azure

**Reply by neelduttahere** [reply]
I think we were not using the latest version of TiTiler, the no-data issue is completely fixed in the latest version. Though it seems I have to pass `asset_as_band` param in URL in order for band expressions to work, which is a welcome change. Thanks for the fast reply!
